Userland/Applications/Presenter/SlideObject.cpp

```diff
@@ -33,13 +33,37 @@ ErrorOr<NonnullRefPtr<SlideObject>> SlideObject::parse_slide_object(JsonObject c
     else
         return Error::from_string_view("Unsupported slide object type"sv);
 
-    slide_object_json.for_each_member([&](auto const& key, auto const& value) {
-        if (key == "type"sv)
+    auto assign_property = [&](auto const& key, auto const& value) {
+        if (key == "type"sv || key == "templates"sv)
             return;
         auto successful = object->set_property(key, value);
         if (!successful)
             dbgln("Storing {:15} = {:20} on slide object type {:8} failed, ignoring.", key, value, type);
-    });
+    };
+
+    // First, assign properties from the templates.
+    auto const& used_templates = slide_object_json.get("templates"sv);
+    if (!used_templates.is_null()) {
+        if (!used_templates.is_array())
+            return Error::from_string_view("Slide object templates are not an array"sv);
+
+        // Most objects have only one template.
+        Vector<DeprecatedString, 1> used_template_ids;
```

Comment:
I think you could `ensure_capacity()` with the size of the used_templates array?